templates/pkg/crd_sdk.go.tpl

```diff
@@ -46,7 +47,13 @@ func (rm *resourceManager) sdkFind(
 		if awsErr, ok := ackerr.AWSError(respErr); ok && awsErr.Code() == "{{ ResourceExceptionCode .CRD 404 }}" {
 			return nil, ackerr.NotFound
 		}
-		return nil, err
+
+		// If there are validation errors in constructed request
+		// treat it as NotFound
```

Comment:
A few things...

First, an invalid parameter isn't a not found :) We should be producing valid Input shapes, so if we're not, let's fix the source of that bug.

Second, we need to handle generic validation of CR input using validating admission webhooks (see #162).

Reply:
I agree to above points and treating ValidationException as NotFound will hide validation-failure before calling Create Api.

I will think of better solution but this was a special scenario. APIGatewayV2 api is a bit different from s3 and ecr. ValidationException occurs even with a legit CR input. See points below.

* To create valid input shape(GetApiInput) in above case, there has to be an existing API(Populated KO status). Which will not happen because we have ReadOne before Create operation.
* S3 and ECR APIs use list operations and filter on a certain field while APIGatewayV2 uses Get(w/ api-id) rather than List.


@jaypipes , I will think of a better solution and  run it by you first before implementing.